### ISSUE TITLE
shake128 test coverage

### ISSUE THREAD

**Issue by inkydragon** [JuliaCrypto/SHA.jl#94]
I'm trying to test those lines:

https://app.codecov.io/gh/JuliaCrypto/SHA.jl/blob/master/src%2Fshake.jl#L84
https://github.com/JuliaCrypto/SHA.jl/blob/88e1c83c32790c1cfbd0d13cc71c7db32bd6a573/src/shake.jl#L74-L76
https://github.com/JuliaCrypto/SHA.jl/blob/88e1c83c32790c1cfbd0d13cc71c7db32bd6a573/src/shake.jl#L83-L88

So I construct an input:

```julia
julia> using SHA
julia> SHA.blocklen(SHAKE_128_CTX) |> Int
168

julia> SHA.shake128(codeunits("0" ^ 167), UInt(32)) |> bytes2hex
"1350314219fb0728c65f0a9f560e70c402d63f1862f9f34555ce27828f5e0373"
julia> SHA.shake128(codeunits("0" ^ 168), UInt(32)) |> bytes2hex
"39ebc9a5df5c81a46d8a6f543326348695ff59d0dd10e020b53cb49a1c1532e4"
```

python output:

```py
>>> import hashlib
>>> h = hashlib.shake_128()
>>> h.update(b'0' * 167)
>>> h.hexdigest(32)
'ff60b0516fb8a3d4032900976e98b5595f57e9d4a88a0e37f7cc5adfa3c47da2'

>>> h = hashlib.shake_128()
>>> h.update(b'0' * 168)
>>> h.hexdigest(32)
'39ebc9a5df5c81a46d8a6f543326348695ff59d0dd10e020b53cb49a1c1532e4'
```

Notice that for the input `"0" ^ 167`, julia's output does not match python's.
There might be a bug here. @immoschuett

**Comment by immoschuett** [comment]
Hi, thanks for the tests @inkydragon 

I also thought about the padding.

I think the same issue is also in the digest for the other SHA3 hash functions.
The padding of julia SHA is not along the NIST standards. 

the Padding in the else case  `usedspace >= blocklen(T)` should be 

`context.buffer[end] = 0x9f` for shake 

and 

`context.buffer[end] = 0x68`  for sha3 digest functions

**Comment by inkydragon** [comment]
Fixed by #95
